src/embed/connection.rs

```diff
@@ -273,15 +273,19 @@ async fn do_put(txn: &RwLock<Transaction<'_>>, req: PutRequest) -> Result<PutRes
     Ok(PutResponse {})
 }
 
-// This fn just forwards to sync::begin_sync at this point. execute() requires its
-// do_ers to take txns even if they don't need it so this fn also helps begin_sync
-// have a simpler signature.
 async fn do_begin_sync<'a, 'b>(
     _: &'a dag::Store,
     _: &'b TxnMap<'a>,
     req: BeginSyncRequest,
 ) -> Result<BeginSyncResponse, sync::BeginSyncError> {
-    let begin_sync_response = sync::begin_sync(&req).await?;
+    // TODO move client up to process() or into a lazy static so we can re-use.
+    let hyper_client_owned: hyper::Client<hyper::client::HttpConnector>;
+    let mut hyper_client_option = None;
```

Comment:
It seems like you can use match to condense this somewhat and avoid the mut option:

```
let hyper_client = match sync::USE_BROWSER_FETCH {
    true => Some(hyper::Client::New()),
    false => None,
}
```

If you then pass `&hyper_client` into `begin_sync` I believe that all the methods that get the thingy out of the option will give it to you by ref too (otherwise there are helpers for that).